### PR TITLE
Swift upgrade

### PR DESCRIPTION
**Pull request by adamfox-nanopay**
Changes to support SMF properties

### CODE REVIEW

**Comment by jlhughes** [comment]
@kgrgreer we are adding these to the 'PropertyInfo'.  Since these are SWIFT/Protocol specific believe we should add this support elsewhere.  Can we use refinements?  java/refinements.

**Comment by adamfox-nanopay** [comment]
> @kgrgreer we are adding these to the 'PropertyInfo'. Since these are SWIFT/Protocol specific believe we should add this support elsewhere. Can we use refinements? java/refinements.

@jlhughes I think we need it on propertyinfo so we can do this type of message construction
```
  properties: [
    {
      name: 'smfMessageType',
      value: 'MT101'
    },
    { class: 'net.nanopay.smf.model.MT20' },
    { class: 'net.nanopay.smf.model.MT21R' },
    { class: 'net.nanopay.smf.model.MT28D' },
    ...
```
Where each property uses these propertyinfo attributes such as 

```
foam.CLASS({
  package: 'net.nanopay.smf.model',
  name: 'MT23E',
  extends: 'foam.core.List',

  properties: [
    [ 'of', 'net.nanopay.smf.model.SMFInstructionCodeDetail' ],
    [ 'name', 'instructionCodeList' ],
    [ 'shortName', net.nanopay.smf.model.SMFPropertyName.MT23E.name ],
    [ 'smfIsRepeatable', true ],
    [ 'smfTag', '23E' ]
  ]
});

```

File: src/foam/java/PropertyInfo.js [hunk 1]
```diff
@@ -155,6 +155,23 @@ foam.CLASS({
       factory: function() { return this.property.sheetsOutput; },
       documentation: 'The sheetsOutput specifies if property shoud be written to Google Sheet on import. eg on Transaction import in case there is Status column transaction\'s status will be written there'
     },
+    {
```
Comment (kgrgreer) [inline src/foam/java/PropertyInfo.js]: Can we refine PropertyInfo from the NP side, rather than introduce this here?

Comment (jlhughes) [reply src/foam/java/PropertyInfo.js]: @adamfox-nanopay see foam/java/refinements.js
Then add your PropertyInfo properties via the refines process on the nanopay side.

**Pull request closed without merging**